scipy/special/cephes/hyp2f1.c

```diff
@@ -590,3 +595,27 @@ static double hyp2f1ra(double a, double b, double c, double x,
 
     return f0;
 }
+
+
+/*
+    15.4.2 Abramowitz & Stegun.
+*/
+static double hyp2f1_neg_c_equal_bc(double a, double b, double x)
+{
+    double k;
+    double err;
+    double collector = 1;
+    double sum = 1;
+    double const max_err = (1e-7/1e-16) - 1;
+
+    for (k = 1; k <= -b; k++) {
+  	collector *= (a + k - 1)*x/k;
+    	sum += collector;
+    
+        if (fabs(collector / sum) > max_err) {
```

Comment:
Please change the error estimate to the form I suggested earlier, so that it's not necessary to make assumptions about the behavior of the sum.

Reply:
I assume that I am wrong here, but I would like to know why: I literally just moved all of the constant parts to one side of the equation and expanded the fabs around the variable expression.

Reply:
This is comparing the current fp error to the current partial sum. This is not necessarily the same as the maximum fp error vs. the final value of the sum.

Consider for example the terms `[1e15, -1e15, 1e30]`. There's no problem computing the value of the sum, but the above if statement bails out. Ruling out that this sort of thing cannot occur in the above sum requires some mathematical argument, and it's simpler to not try doing that.